lib/removePipelineExtras.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+
+module.exports = removePipelineExtras;
+
+function removePipelineExtras(gltf) {

```

Comment:
Ah, you are right.  So we would still use an approach like `clone`, but we need to check each object to see if it is one of the objects that the glTF spec says can have an `extras` as you suggested above..